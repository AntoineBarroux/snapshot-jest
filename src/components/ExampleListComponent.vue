<template>
<!--  <div>-->
<!--    <p>Nombre d'éléments : {{ numberOfElements }}</p>-->

    <ul>
      <li v-for="item in items" :key="item.id">
        {{item.value}}
      </li>
    </ul>
<!--  </div>-->
</template>

<script>
    import {mapActions, mapGetters} from "vuex";

    export default {
        name: 'ExampleListComponent',
        created() {
            this.loadItems();
        },
        methods: {
            ...mapActions({
                loadItems: 'exampleStore/loadItems'
            })
        },
        computed: {
            ...mapGetters({
                items: 'exampleStore/items'
            }),
            numberOfElements() {
              return this.items.length;
            }
        }
    }
</script>

<style>

</style>
