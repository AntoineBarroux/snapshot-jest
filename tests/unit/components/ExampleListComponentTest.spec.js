import {createLocalVue, shallowMount} from "@vue/test-utils";
import Vuex from "vuex";
import ExampleListComponent from "../../../src/components/ExampleListComponent";

const localVue = createLocalVue();
localVue.use(Vuex);

describe('', () => {

    let store;
    const elements = [
        {
            id: '1',
            value: 'Premier élément'
        },
        {
            id: '2',
            value: 'Deuxieme élément'
        }
    ];

    beforeEach(() => {
        store = new Vuex.Store({
            modules: {
                exampleStore: {
                    namespaced: true,
                    state: {
                        items: elements
                    },
                    actions: {
                        loadItems: jest.fn(() => Promise.resolve(elements)),
                    },
                    getters: {
                        'items': state => state.items
                    }
                }
            }
        });
    });

    it('Charge les données à la création du composant', (done) => {
        store.dispatch = jest.fn(() => Promise.resolve());
        const wrapper = shallowMount(ExampleListComponent, {
            localVue, store
        });

        wrapper.vm.$nextTick().then(() => {
            expect(store.dispatch).toHaveBeenCalledWith('exampleStore/loadItems');
            done();
        });
    });

    it('S\'affiche correctement', () => {
        const wrapper = shallowMount(ExampleListComponent, {
            localVue, store
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
