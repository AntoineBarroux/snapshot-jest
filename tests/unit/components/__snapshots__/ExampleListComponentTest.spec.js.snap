// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`S'affiche correctement 1`] = `
<div>
  <p>
    Nombre d'éléments : 2
  </p>
   
  <ul>
    <li>
      
      Premier élément
    
    </li>
    <li>
      
      Deuxieme élément
    
    </li>
  </ul>
</div>
`;
